### PR TITLE
Models

### PR DESCRIPTION
**Pull request by codeMonkeyHopeful**
added model frameworks

### CODE REVIEW

File: server/database/index.js [hunk 1]
```diff
@@ -5,11 +5,16 @@ const Group = require('./models/Group');
 const Chore = require('./models/Chore');
 const Trade = require('./models/Trade');
 const Vote = require('./models/Vote');
+const EthereumWallet = require('./models/EthereumWallet')
+const UserGroup = require('./models/UserGroup')
+const AssignedChores = require('./models/AssignedChores')
```
Comment (HypeDis) [inline server/database/index.js]: why plural?

File: server/database/models/EthereumWallet.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const Sequelize = require('./node_modules/sequelize');
+const db = require('../db');
+
+const EthereumWallet = db.define('ethereumWallet', {
+    userId: {
```
Comment (HypeDis) [inline server/database/models/EthereumWallet.js]: userId is made through associations

File: server/database/models/AssignedChores.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const Sequelize = require('./node_modules/sequelize');
+const db = require('../db');
+
+const AssignedChores = db.define('assignedChores', {
+    assignedChoreId: {
+        type: Sequelize.INTEGER,
+        allowNull: false,
+    },
+    userId: {
```
Comment (HypeDis) [inline server/database/models/AssignedChores.js]: userId and choreId will be made through associations

File: server/database/models/AssignedChores.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+const Sequelize = require('./node_modules/sequelize');
+const db = require('../db');
+
+const AssignedChores = db.define('assignedChores', {
+    assignedChoreId: {
```
Comment (HypeDis) [inline server/database/models/AssignedChores.js]: assignedChore.id is created when table is created

File: server/database/models/SellChore.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+const Sequelize = require('sequelize')
+const db = require('../db')
+
+const SellChore = db.define('sellChore', {
+    saleId: {
```
Comment (HypeDis) [inline server/database/models/SellChore.js]: saleId is made on table creation

File: server/database/models/SellChore.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        type: Sequelize.INTEGER,
+        allowNull: false,
+    },
+    originalOwnerId: {
```
Comment (HypeDis) [inline server/database/models/SellChore.js]: originalOwnerId, newOwnerId, and assignedChoreId are created through associations

File: server/database/models/SwapChore.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
+const Sequelize = require('sequelize')
+const db = require('../db')
+
+const SwapChore = db.define('swapChore', {
+    swapId: {
+        type: Sequelize.INTEGER,
+        allowNull:false,
+    },
+    user1: {
```
Comment (HypeDis) [inline server/database/models/SwapChore.js]: user1  ---> assignedChoreId2 made through associations

File: server/database/index.js [hunk 8]
```diff
@@ -5,11 +5,16 @@ const Group = require('./models/Group');
...
+const SellChore = require('./models/SellChore')
+const SwapChore = require('./models/SwapChore')
 
 // add model associations here
 
 /*----- User and Group Associations -----*/
-User.belongsToMany(Group, { through: 'UserGroup' });
```
Comment (HypeDis) [inline server/database/index.js:12]: good catch

**Review by HypeDis — CHANGES_REQUESTED** [review]
see comments

**Comment by codeMonkeyHopeful** [comment]
Updated to pull out associations in the models themselves

File: server/database/index.js (hunk 1 above)
Comment (codeMonkeyHopeful) [reply server/database/index.js]: made table name singular

File: server/database/db.js [hunk 9]
```diff
@@ -3,6 +3,6 @@ if (process.env.NODE_ENV !== 'production') {
 }
 
 const Sequelize = require('sequelize');
-const db = new Sequelize(process.env.DATABASE_URL, { logging: false });
+const db = new Sequelize(/*process.env.DATABASE_URL*/'postgres://:5432/choreApp', { logging: false });
```
Comment (HypeDis) [inline server/database/db.js]: change this back

File: server/database/models/AssignedChore.js [hunk 10]
```diff
@@ -1,5 +1,5 @@
-const Sequelize = require('./node_modules/sequelize');
```
Comment (HypeDis) [inline server/database/models/AssignedChore.js]: whats going on here?

File: server/database/models/UserGroup.js [hunk 11]
```diff
@@ -1,5 +1,5 @@
-const Sequelize = require('./node_modules/sequelize');
```
Comment (HypeDis) [inline server/database/models/UserGroup.js]: delete this file

File: server/database/index.js [hunk 12]
```diff
@@ -35,6 +35,35 @@ User.hasMany(Trade, { foreignKey: 'newOwnerId' });
...
+SellChore.belongsTo(Chore, {as: 'assignedChore'})
+
+//User.hasMany(SellChore)
+// SellChore.belongsTo(Chore)
+// Chore.hasMany(SellChore)
+
+/*-------SwapChore Associations------*/
+SwapChore.belongsTo(User, {as: 'user1'})
+SwapChore.belongsTo(User, {as: 'user2'})
+SwapChore.belongsTo(AssignedChore, {as: 'assignedChoreId1'})
```
Comment (HypeDis) [inline server/database/index.js]: assignedChore1 and assignedChore2, drop the 'Id'

**Review by HypeDis — CHANGES_REQUESTED** [review]

File: server/database/db.js (hunk 9 above)
Comment (codeMonkeyHopeful) [reply server/database/db.js]: my bad, i must embrace the .env

File: server/database/models/AssignedChore.js (hunk 10 above)
Comment (codeMonkeyHopeful) [reply server/database/models/AssignedChore.js]: looks like this was in a previous commit, not sure why its showing up.  Will double chekc the push this time to make sure it goes through

File: server/database/models/UserGroup.js (hunk 11 above)
Comment (codeMonkeyHopeful) [reply server/database/models/UserGroup.js]: same as above

File: server/database/index.js (hunk 12 above)
Comment (codeMonkeyHopeful) [reply server/database/index.js]: dropped our the 'Id' portion of header name

File: server/database/db.js (hunk 9 above)
Comment (codeMonkeyHopeful) [reply server/database/db.js]: fixed

File: server/database/models/AssignedChores.js (hunk 4 above)
Comment (codeMonkeyHopeful) [reply server/database/models/AssignedChores.js]: updated with associations

File: server/database/models/SellChore.js (hunk 5 above)
Comment (codeMonkeyHopeful) [reply server/database/models/SellChore.js]: associations created

File: server/database/models/SellChore.js (hunk 6 above)
Comment (codeMonkeyHopeful) [reply server/database/models/SellChore.js]: same

File: server/database/models/SwapChore.js (hunk 7 above)
Comment (codeMonkeyHopeful) [reply server/database/models/SwapChore.js]: associations made

**Review by codeMonkeyHopeful — COMMENTED** [review]
done

File: server/database/index.js [hunk 13]
```diff
@@ -19,17 +24,46 @@ Group.hasMany(Chore);
 /*----- Vote Associations -----*/
 User.hasMany(Vote, { foreignKey: 'voterId', sourceKey: 'id' });
 Vote.belongsTo(User, { foreignKey: 'voterId', targetKey: 'id' });
-// Vote.belongsTo(AssignedChore)
+// Vote.belongsTo(AssignedChores)
```
Comment (HypeDis) [inline server/database/index.js:27]: Singular

File: server/database/index.js [hunk 14]
```diff
@@ -19,17 +24,46 @@ Group.hasMany(Chore);
...
 // AssignedChore.hasMany(Vote);
 
 /*----- Trade Associations -----*/
 Trade.belongsTo(User, { foreignKey: 'originalOwnerId' });
 Trade.belongsTo(User, { foreignKey: 'newOwnerId' });
 User.hasMany(Trade, { foreignKey: 'originalOwnerId' });
 User.hasMany(Trade, { foreignKey: 'newOwnerId' });
-// Trade.belongsTo(AssignedChore)
+// Trade.belongsTo(AssignedChores)
```
Comment (HypeDis) [inline server/database/index.js]: Singular

File: server/database/models/AssignedChore.js [hunk 15]
```diff
@@ -0,0 +1,15 @@
+const Sequelize = require('sequelize');
+const db = require('./../db');
+
+const AssignedChore = db.define('assignedChores', {
```
Comment (HypeDis) [inline server/database/models/AssignedChore.js]: 'assignedChore'

**Review by HypeDis — CHANGES_REQUESTED** [review]

File: server/database/index.js (hunk 14 above)
Comment (codeMonkeyHopeful) [reply server/database/index.js]: any reason we want to keep these commented out code?  Fixed in the meantime.

File: server/database/models/AssignedChore.js (hunk 15 above)
Comment (codeMonkeyHopeful) [reply server/database/models/AssignedChore.js]: good catch, completely spaced over this one.

**Review by HypeDis — APPROVED** [review]

**Pull request merged**